src/main/java/com/cloudbees/jenkins/support/util/WordReplacer.java

```diff
@@ -0,0 +1,168 @@
+     */
+    public String replaceWordIgnoreCase(String input, String word, String replace) {
+        StringBuilder sb = new StringBuilder(input);
+        replaceWordIgnoreCase(sb, word, replace);
+        return sb.toString();
+    }
+
+    /**
+     * Replace all occurrences of word by replace in the input. The replacement happens only if the word to replace is
+     * separated from others words, that is, it's not part of a word. The implementation is that the previous and next
+     * characters of the word must be a {@link Character#isLetterOrDigit(char)} char.
+     * @param input text where the replacements take place
+     * @param word the word to replace
+     * @param replace the new text to use
+     */
+    public static void replaceWord(StringBuilder input, String word, String replace) {
+        replaceWord(input, word, replace, false);
+    }
+
+
+    public static void replaceWordIgnoreCase(StringBuilder input, String word, String replace ) {
+        replaceWord(input, word, replace, true);
+    }
+
+    // Main method
+    private static void replaceWord(StringBuilder input, String word, String replace, boolean ignoreCase) {
+        // The string where we look for
+        StringBuilder workInput;
+        // The string to find (lower case if ignoreCase)
+        String workWord = word;
+
+        // Use the lowercase versions to compare, but replace in the original input
+        if (ignoreCase) {
+            workInput = new StringBuilder(input.toString().toLowerCase(Locale.ENGLISH));
+            workWord = word.toLowerCase(Locale.ENGLISH);
+        } else {
+            workInput = new StringBuilder(input);
+        }
+
+        int pos = 0;
+        do {
+            // Search for the word but from the previous found index, to avoid infinite loop if the replace word
+            // contains the word
+            pos = workInput.indexOf(workWord, pos);
+
+            // If word is not found
+            if (pos == -1) {
+                break;
+            }
+
+            // If the previous char is a word letter, don't replace, look for the next one.
+            if (pos > 0) {
+                char prevChar = workInput.charAt(pos - 1);
+                if (Character.isLetterOrDigit(prevChar)) {
+                    pos = pos + workWord.length();
+                    continue;
+                }
+            }
+
+            // If the next char is a word letter, don't replace, look for the next
+            int nextPos = pos + workWord.length();
+            if (nextPos < workInput.length()) {
+                char nextChar = workInput.charAt(nextPos);
+                if (Character.isLetterOrDigit(nextChar)) {
+                    pos = pos + workWord.length();
+                    continue;
+                }
+            }
+
+            // replace this word but in the original input (without changing the case)
+            input.replace(pos, pos + workWord.length(), replace);
```

Comment:
Isn't the replacement assigned anywhere?

Reply:
input is a **StringBuilder**, so it's changed itself with the replace method.